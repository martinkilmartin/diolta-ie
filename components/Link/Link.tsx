type Props = {
  text: string
  href?: string
}

const Link = ({ text, href = '#' }: Props): JSX.Element => (
  <a
    className="transition-colors duration-200 hover:text-accent-focus"
    href={href}
  >
    {text}
  </a>
)

export default Link
