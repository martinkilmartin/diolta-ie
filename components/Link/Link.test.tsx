import renderer from 'react-test-renderer'
import Link from './Link'

test('Link displays text', () => {
  const linkText = 'Click Me!'
  const linkRenderer = renderer.create(<Link text={linkText} />)
  const tree = linkRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const linkInstance = linkRenderer.root
  expect(
    linkInstance.findByProps({
      className: 'transition-colors duration-200 hover:text-accent-focus',
    }).children[0]
  ).toEqual(linkText)
})
