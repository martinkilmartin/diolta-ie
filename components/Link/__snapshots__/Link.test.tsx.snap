// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link displays text 1`] = `
<a
  className="transition-colors duration-200 hover:text-accent-focus"
  href="#"
>
  Click Me!
</a>
`;
